package-system/OpenSSL/FindOpenSSL.cmake.template

```diff
@@ -6,29 +6,64 @@
 #
 #
 
-set(TARGET_WITH_NAMESPACE "3rdParty::OpenSSL")
-if (TARGET $${TARGET_WITH_NAMESPACE})
+set(OPENSSL_O3DE_NAMESPACE "3rdParty::OpenSSL")
+if (TARGET $${OPENSSL_O3DE_NAMESPACE})
     return()
 endif()
 
+set(SSL_TARGETNAME "OpenSSL::SSL")
+set(CRYPTO_TARGETNAME "OpenSSL::Crypto")
+
+# we're trying to be a drop-in replacement for the FindOpenSSL.cmake that is shipped
+# with CMake itself, so we set the same variables with the same uppercase for compatibility
+# for questions about these variables, see https://cmake.org/cmake/help/latest/module/FindOpenSSL.html
+set(OPENSSL_FOUND True)
 set(OPENSSL_INCLUDE_DIR $${CMAKE_CURRENT_LIST_DIR}/OpenSSL/include)
 set(OPENSSL_LIBS_DIR $${CMAKE_CURRENT_LIST_DIR}/OpenSSL/$$<$$<CONFIG:debug>:debug/>lib)
+set(OPENSSL_CRYPTO_LIBRARY $${OPENSSL_LIBS_DIR}/libcrypto$${CMAKE_STATIC_LIBRARY_SUFFIX})
+set(OPENSSL_CRYPTO_LIBRARIES
+    $${OPENSSL_CRYPTO_LIBRARY}
+    ${CRYPTO_LIBRARY_DEPENDENCIES})
+set(OPENSSL_SSL_LIBRARY $${OPENSSL_LIBS_DIR}/libssl$${CMAKE_STATIC_LIBRARY_SUFFIX})
+set(OPENSSL_SSL_LIBRARIES $${OPENSSL_SSL_LIBRARY})
+set(OPENSSL_LIBRARIES
+    $${OPENSSL_CRYPTO_LIBRARIES}
+    $${OPENSSL_SSL_LIBRARIES})
+set(OPENSSL_VERSION_STRING "1.1.1m")
+set(OPENSSL_VERSION $${OPENSSL_VERSION_STRING})
 
-set(OPENSSL_COMPILE_DEFINITIONS
-    OPENSSL_ENABLED
-${CUSTOM_ADDITIONAL_COMPILE_DEFINITIONS})
-
-set(OPENSSL_LIBRARY
-    "$${OPENSSL_LIBS_DIR}/libssl$${CMAKE_STATIC_LIBRARY_SUFFIX}"
-    "$${OPENSSL_LIBS_DIR}/libcrypto$${CMAKE_STATIC_LIBRARY_SUFFIX}"
-${CUSTOM_ADDITIONAL_LIBRARIES})
+add_library($${SSL_TARGETNAME} STATIC IMPORTED GLOBAL)
+set_target_properties($${SSL_TARGETNAME} PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES "C")
+set_target_properties($${SSL_TARGETNAME} PROPERTIES IMPORTED_LOCATION "$${OPENSSL_CRYPTO_LIBRARY}")
```

Comment:
Why does the ssl imported target use the crypto library?
I would have expected the ssl imported target use the ssl library, and the crypto imported target to use the crypto library.